### PR TITLE
VFDs, maps, event sets. Done?

### PR DESCRIPTION
**Pull request by gheber**
Added missing VFD calls, map and event set modules.

### CODE REVIEW

File: src/H5FDcore.h [hunk 1]
```diff
@@ -25,8 +25,70 @@
...
-H5_DLL hid_t  H5FD_core_init(void);
+H5_DLL hid_t H5FD_core_init(void);
+
+/**
+ * \ingroup FAPL
+ *
+ * \brief Modifies the file access property list to use the #H5FD_CORE driver
+ *
+ * \fapl_id
+ * \param[out] increment Size, in bytes, of memory increments
```
Comment (ChristopherHogan) [inline src/H5FDcore.h]: According to the current RM, this is an `in` param.

File: src/H5FDcore.h [hunk 2]
```diff
@@ -25,8 +25,70 @@
...
+ * \param[in] backing_store Size, in bytes, of memory increments
+ *             contents to disk when the file is closed
```
Comment (ChristopherHogan) [inline src/H5FDcore.h]: This description is different from the current RM: "Boolean flag indicating whether to write the file contents to disk when the file is closed."

File: doxygen/dox/ReferenceManual.dox [hunk 3]
```diff
@@ -1,5 +1,8 @@
 /** \page RM Reference Manual
 
+\todo Add the Event Sets module
+\todo Add the Map module
+
 The functions provides by the HDF5 C-API are grouped into the following
```
Comment (ChristopherHogan) [inline doxygen/dox/ReferenceManual.dox]: I know this line isn't part of the PR, but I happened to notice a typo for "provided" here.

File: src/H5FDdirect.h [hunk 4]
```diff
@@ -37,8 +37,69 @@ extern "C" {
...
-H5_DLL hid_t  H5FD_direct_init(void);
+H5_DLL hid_t H5FD_direct_init(void);
+
+/**
+ * \ingroup FAPL
+ *
+ * \brief Sets up use of the direct I/O driver
+ *
+ * \fapl_id
+ * \param[in] boundary Required memory alignment boundary
```
Comment (ChristopherHogan) [inline src/H5FDdirect.h]: This parameter should be called "alignment," right?

File: src/H5FDdirect.h [hunk 5]
```diff
@@ -37,8 +37,69 @@ extern "C" {
...
+ * \brief Retrieves direct I/O driver settings
+ *
+ * \fapl_id
+ * \param[out] boundary Required memory alignment boundary
+ * \param[out] block_size File system block size
+ * \param[out] cbuf_size Copy buffer size
+ * \returns \herr_t
+ *
+ * \details H5Pget_fapl_direct() retrieves the required memory alignment (\p
+ *          alignment), file system block size (\p block_size), and copy buffer
```
Comment (ChristopherHogan) [inline src/H5FDdirect.h:93]: This line references an `alignment` parameter, but the function names the parameter `boundary`.

File: src/H5ESpublic.h [hunk 6]
```diff
@@ -119,14 +123,157 @@ How to Trace Async Operations?
 extern "C" {
 #endif
 
-H5_DLL hid_t  H5EScreate(void);
+/**
+ * \ingroup H5ES
+ *
+ * \brief Creates an event set
+ *
+ * \returns \hid_ti{event set}
```
Comment (jya-kmu) [inline src/H5ESpublic.h:131]: Typo "hid_ti"?

File: src/H5FDmpio.h [hunk 7]
```diff
@@ -44,14 +44,237 @@ H5_DLLVAR hbool_t H5FD_mpi_opt_types_g;
...
+ *          HDF5 library normally uses an internal algorithm to determine
+ *          whether that I/O activity should be conducted as one linked-chunk
+ *          I/O or as multi-chunk I/O. H5Pset_dxpl_mpio_chunk_opt() is provided
+ *          so that an application can override the library's algorithm in
+ *          circumstances where the library might lack the information needed to
+ *          make an optimal decision.
+ *
+ *          H5Pset_dxpl_mpio_chunk_opt() works by setting one of the following
+ *          flags in the parameter \p opt_mode:
+ *          - #H5FD_MPIO_CHUNK_ONE_IO -	Do one link chunked I/O
```
Comment (jya-kmu) [inline src/H5FDmpio.h]: Alignment seems off from "Do one link..."

**Review by ChristopherHogan — APPROVED** [review]

File: doxygen/dox/ReferenceManual.dox (hunk 3 above)
Comment (epourmal) [reply doxygen/dox/ReferenceManual.dox]: Nice catch :-)

**Review by byrnHDF — APPROVED** [review]

File: src/H5ESpublic.h (hunk 6 above)
Comment (gheber) [reply src/H5ESpublic.h:131]: No. I want the text to expand into "Returns _an_ event set identifier..."

File: src/H5FDmpio.h (hunk 7 above)
Comment (gheber) [reply src/H5FDmpio.h]: I agree, it looks a little out-of-kilter because it's a bulleted list and not a table.

File: src/H5FDcore.h [hunk 8]
```diff
@@ -25,8 +25,70 @@
...
+ *          be increased each time more memory is required.
+ *
+ *          While using H5Fcreate() to create a core file, if the \p
+ *          backing_store is set to 1 (TRUE), the file contents are flushed to a
+ *          file with the same name as this core file when the file is closed or
+ *          access to the file is terminated in memory.
+ *
+ *          The application is allowed to open an existing file with #H5FD_CORE
+ *          driver. While using H5Fopen() to open an existing file, if the \p
+ *          backing_store is set to 1 and the \c flags for H5Fopen() is set to
```
Comment (bmribler) [inline src/H5FDcore.h]: 1 (TRUE)?

File: src/H5FDcore.h [hunk 9]
```diff
@@ -25,8 +25,70 @@
...
+ *          #H5F_ACC_RDWR, any change to the file contents are saved to the file
+ *          when the file is closed. If \p backing_store is set to 0 and the \c
```
Comment (bmribler) [inline src/H5FDcore.h]: 0 (FALSE)?

File: src/H5FDmulti.h [hunk 10]
```diff
@@ -25,11 +25,228 @@
...
+ *          The array \p memb_name should be a name generator (a
+ *          \Code{printf}-style format with a \Code{%s} which will be replaced
+ *          with the name passed to H5FDopen(), usually from H5Fcreate() or
+ *          H5Fopen()).
+ *
+ *          The array \p memb_addr specifies the offsets within the virtual
+ *          address space, from 0 (zero) to #HADDR_MAX, at which each type of
+ *          data storage begins.
+ *
+ *          If \p relax is set to TRUE (or 1), then opening an existing file for
```
Comment (bmribler) [inline src/H5FDmulti.h]: Should "TRUE (or 1)" or "1 (TRUE)" be used consistently throughout?  I think we should follow the existing style.

File: src/H5Fmodule.h [hunk 11]
```diff
@@ -31,8 +31,34 @@
...
  * \defgroup H5F H5F
- * \brief File Interface
- * \todo Describe concisely what the functions in this module are about.
+ *
+ * Use the functions in this module to manage HDF5 files.
+ *
+ * In the code snippets below, we show the skeletal life cycle of an HDF5 file,
+ * when creating a new file (left) or when opening an existing file (right).
+ * File creation is essentially controlled through \ref FCPL, and file access to
+ * new and existing files is controlled through \ref FAPL. The file \c name and
```
Comment (bmribler) [inline src/H5Fmodule.h:40]: The use of CAP and no cap for variable names is inconsistent.

File: src/H5Mpublic.h [hunk 12]
```diff
@@ -81,38 +85,397 @@ extern "C" {
...
                              const char *name, hid_t key_type_id, hid_t val_type_id, hid_t lcpl_id,
                              hid_t mcpl_id, hid_t mapl_id, hid_t es_id);
+
+/**
+ * \ingroup H5M
+ *
+ * \brief
+ *
+ * \details
+ *
```
Comment (bmribler) [inline src/H5Mpublic.h:129]: Just want to make sure that this is not unintentionally left blank...

File: src/H5Rpublic.h [hunk 13]
```diff
@@ -74,7 +74,7 @@ typedef haddr_t hobj_ref_t;
 /**
  * Buffer to store heap ID and index
  *
- * This needs to be large enough to store largest \ref haddr_t in a worst case
+ * This needs to be large enough to store largest haddr_t in a worst case
```
Comment (bmribler) [inline src/H5Rpublic.h]: Does haddr_t need a format code?

**Review by bmribler — APPROVED** [review]

File: src/H5Fmodule.h (hunk 11 above)
Comment (gheber) [reply src/H5Fmodule.h:40]: These (FCPL, FAPL) are NOT variable names.

File: src/H5Mpublic.h (hunk 12 above)
Comment (gheber) [reply src/H5Mpublic.h:129]: Correct. Intentional for now.

File: src/H5Rpublic.h (hunk 13 above)
Comment (gheber) [reply src/H5Rpublic.h]: Corrected to `#haddr_t`.

**Pull request merged**